recipes/cpp_project_framework/all/test_package/CMakeLists.txt

```
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 3.1)
+project(PackageTest CXX)
+
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED True)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup()
```

```suggestion
conan_basic_setup(TARGETS)
```
Otherwise, the `cmake` generator will mask `cmake_find_package`

---

actually I had tested to change it to `TARGETS` locally but have problem building the test package (cannot find my project's header files) while no issue without it

---

So it's broken. If I consume this package, I have the choice of using targets too. Which error did you have, let's try to fix it.

---

```
  Checking Build System
  Building Custom Rule conan-center-index/recipes/cpp_project_framework/all/test_package/CMakeLists.txt
  test_package.cpp
conan-center-index\recipes\cpp_project_framework\all\test_package\test_package.cpp(1,10): error C1083:  : 'cpp_project_framework/cpp_project_framework.h': No such file or directory [conan-center-index\recipes\c
pp_project_framework\all\test_package\build\ecbfb49a4e2ea8618319be5f9e02c8e0621391fb\test_package.vcxproj]
ERROR: cpp_project_framework/1.0.0 (test package): Error in build() method, line 15
        cmake.build()
        ConanException: Error 1 while executing cmake --build conan-center-index\recipes\cpp_project_framework\all\test_package\build\ecbfb49a4e2ea8618319be5f9e02c8e0621391fb --config Release -- /m:12 /verbosity:minimal
```

---

Okay, I`ll try to reproduce locally. Thank you for sharing!

---

Fixed with my new comments. Keep `TARGETS` declared.

---

changed in latest version